packages/ns-api/files/ns.backup

```diff
@@ -129,7 +129,7 @@ elif cmd == 'call':
             # reboot takes a few seconds to complete, enough to send the response
             print(json.dumps({'message': 'success'}))
         except RuntimeError as error:
-            print(json.dumps(utils.generic_error(error.args[0])))
+            print(json.dumps(utils.validation_error('passphraseRef',error.args[0] )))
```

Comment:
- `restore failed` is a runtime error and should be returned to the UI as `generic_error`
- `decryption failed` is raised when the backup file is invalid or the passphrase is wrong (more likely): let's always handle it as a validation error
- Let's put a `if` to check the error string contained in the runtime error to return to the UI a generic error for `restore failed` and a validation error for `decrypt failed`
- `ref`s are a Vue concept, let's use `passphrase` instead of `passphraseRef`

Reply:
There are 2 exceptions that can be catched:
- `decryption failed` coming from `restore_backup`
- `restore failed` still coming from `restore_backup`

I'd suggest the following (and Giacomo is gonna hate me)
- create two different exceptions in the `restore_backup`, extending the RuntimeException one (just a different name will suffice)
- return `generic_error` or `validation error` using `except` (while checking the content of an exception is fine, you will never know that the exception might contain...)
- add a final `except Exception` to catch an unforseen error coming from wherever.

Reply:
Don't agree, if the user puts in a random file the issue is not the passphrase, is the file

Reply:
Okay, final decision from the PM, use the `validation_error` method to send the error to validate to the user
I'd stick with what Andrea said, just a `passphrase` will suffice
Everything will be taken care of from frontend side